src/isDeepEqual.ts

```diff
@@ -64,6 +64,29 @@ export function isDeepEqual(): unknown {
   return purry(isDeepEqualImplementation, arguments);
 }
 
+function areMapsEqual<T extends Map<unknown, unknown>>(
+  data: T,
+  other: T,
+): boolean {
```

Comment:
1. Name can be consistent with the name of the main function, `isMapsDeepEqual` or `isDeepEqualMaps`. Although the english is broken, it's the same "brokenness" of isDeepEqual (because it can also be named areDeepEqual)... in general, functions returning boolean should be named isXXX.
2. You don't need the generic here, it doesn't add anything to the typing of the function.
```suggestion
function isDeepEqualMaps(
  data: Map<unknown, unknown>,
  other: Map<unknown, unknown>,
): boolean {
```